### PR TITLE
Mutable Provider Documentation

### PR DESCRIPTION
**Pull request by epps**
# Description

This PR updates the documentation on providers with a list of each provider's updatable configuration fields.

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: docs/providers/azure.md [hunk 1]
```diff
@@ -32,11 +34,18 @@ featureform apply azure_blob_config.py --host $FEATUREFORM_HOST
 
 We can re-verify that the provider is created by checking the [Providers tab of the Feature Registry](../getting-started/exploring-the-feature-registry.md).
 
+### Updatable Configuration Fields
+
+* `description`
+* `account_name`
```
Comment (sdreyer) [inline docs/providers/azure.md:40]: Believe this should be immutable

File: docs/providers/spark_databricks.md [hunk 2]
```diff
@@ -51,11 +53,22 @@ spark = ff.register_spark(
...
+
 {% endcode %}
 
+### Updatable Configuration Fields
+
+* `description`
+* `username` (Executor)
+* `password` (Executor)
+* `token` (Executor)
+* `account_name` (File Store)
```
Comment (sdreyer) [inline docs/providers/spark_databricks.md:65]: Believe account name is immutable

**Review by sdreyer — APPROVED** [review]

File: docs/getting-started/registering-infrastructure-providers.md [hunk 3]
```diff
@@ -61,6 +63,12 @@ Then, we'll use the Featureform CLI to register them
 featureform apply providers.py
 ```
 
+## Updating Providers
+
+In the event a provider's credentials change, Featureform supports updating previously registered providers' configurations. To update a provider, simply make the necessary changes to `providers.py`, for example, and register them again using the CLI command.
```
Comment (MMBazel) [inline docs/getting-started/registering-infrastructure-providers.md:68]: Nitpick: Language feels a bit clunky propose simplifying it, maybe like: 

"To update a previously registered provider's configuration, make the necessary changes to providers.py (for example) and register them again using the CLI command. 

The description field can always be updated, but which configuration fields are updatable depends on the provider type. "

Not sure if that flows better?

You can find the list of valid fields for each provider type on their specific page below.

File: docs/providers/cassandra.md [hunk 4]
```diff
@@ -34,3 +34,11 @@ featureform apply cassandra_config.py --host $FEATUREFORM_HOST
 ```
 
 We can re-verify that the provider is created by checking the [Providers tab of the Feature Registry](../getting-started/exploring-the-feature-registry.md).
+
+### Updatable Configuration Fields
```
Comment (MMBazel) [inline docs/providers/cassandra.md:38]: I wonder if we'd want to call them "Mutable" -- I know "Updatable" is more accurate but it also feels clunky.

File: docs/providers/kubernetes.md [hunk 5]
```diff
@@ -41,12 +43,21 @@ k8s_store = ff.register_k8s(
...
 )
 ```
+
 {% endcode %}
 
+### Updatable Configuration Fields
+
+* `description`
+* `docker_image`
+* File store updatable configuration fields
```
Comment (MMBazel) [inline docs/providers/kubernetes.md:53]: Should this be deleted? "* File store updatable configuration fields"

**Review by MMBazel — COMMENTED** [review]
Docs look good to me! 

I left some comments on syntax (thanks for catching some spelling errors on my part!).

File: docs/providers/kubernetes.md (hunk 5 above)
Comment (epps) [reply docs/providers/kubernetes.md:53]: I intended this to mean, "Whichever fields are mutable in your preferred file store"

**Pull request merged**